zerver/tests/test_subs.py

```diff
@@ -3013,9 +3012,6 @@ def test_successful_subscriptions_add_with_announce(self) -> None:
         self.test_realm.notifications_stream_id = notifications_stream.id
         self.test_realm.save()
 
-        # Delete the UserProfile from the cache so the realm change will be
-        # picked up
-        cache.cache_delete(cache.user_profile_by_email_cache_key(self.test_email))
```

Comment:
I suspect these were probably already unnecessary, because `realm.save()` should be flushing this anyway, right?

Reply:
That's right. I extracted this into a separate commit for clarity